### ISSUE TITLE
bug: @RequestHeader/@RequestHeader/@InputArgument doesn't respect name alias and default value

### ISSUE THREAD

**Issue by iuliiasobolevska** [Netflix/dgs-framework#200]
* If you use the "name" field of the RequestHeader annotation, it will not be read. Spring uses them as Alias' for each other, but DGS looks at [the value only](https://github.com/Netflix/dgs-framework/blob/4e7b96b621243d994e3c99e7553ef7bd23979760/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt#L307).
* Similarly, DGS doesn't respect the defaultValue.

Example:
```
...
@RequestHeader(value="User-Agent", defaultValue="some example user agent value") final String userAgent
...
```

**Comment by paulbakker** [comment]
Good catch!

**Comment by csabakos** [comment]
@paulbakker It looks like DGS still doesn't respect `defaultValue`.

Also it's worth pointing out that it doesn't respect `required` either, although that wasn't mentioned in this ticket.

**Comment by paulbakker** [comment]
It's not released yet, so did you test with a locally build snapshot?

**Comment by csabakos** [comment]
> It's not released yet, so did you test with a locally build snapshot?

Yes, I tested with the current master (commit abc57436766486d93cd82497fb8fda9341eb4fbc)

**Comment by paulbakker** [comment]
I understand the issue now. What's fixed is the `default value` (a value without a name), but not the `defaultValue` argument. I hadn't even looked at the latter and wasn't aware of it being an issue, so thanks for reporting that.

I'll create a separate issue for this for clarity.

**Comment by csabakos** [comment]
Awesome, thank you @paulbakker!
